core/src/main/java/org/apache/iceberg/avro/AvroSchemaVisitor.java

```diff
@@ -103,6 +107,10 @@ public T map(Schema map, T value) {
     return null;
   }
 
+  public T variant(Schema variant) {
+    throw new UnsupportedOperationException("Unsupported type: variant");
```

Comment:
I think that this needs to visit the metadata and value fields within the variant, similar to how map key and value are visited.